### PR TITLE
planner: update docs related to plan cache

### PR DESCRIPTION
**Pull request by qw4990**
### What is changed, added or deleted? (Required)
Update document to add plan cache diagnostics and some examples for v6.6

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by fzzf678 — APPROVED** [review]

File: sql-prepared-plan-cache.md [hunk 1]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
 1 row in set (0.00 sec)
 ```
 
+## Prepared Plan Cache 诊断
+
+对于无法进行缓存的查询或计划，TiDB 会通过 warning 的方式输出其无法被缓存的原因。
+
+```
```
Comment (TomShawn) [inline sql-prepared-plan-cache.md:139]:
````suggestion
```sql
````

File: sql-prepared-plan-cache.md [hunk 2]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
 1 row in set (0.00 sec)
 ```
 
+## Prepared Plan Cache 诊断
```
Comment (TomShawn) [inline sql-prepared-plan-cache.md:135]:
```suggestion
## 诊断 Prepared Plan Cache
```

File: sql-prepared-plan-cache.md [hunk 3]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
...
+
+对于无法进行缓存的查询或计划，TiDB 会通过 warning 的方式输出其无法被缓存的原因。
```
Comment (TomShawn) [inline sql-prepared-plan-cache.md:137]:
```suggestion
对于无法进行缓存的查询或计划，可通过 `SHOW WARNINGS` 语句查看查询或计划是否被缓存。如果未被缓存，则可在结果中查看无法被缓存的原因。示例如下：
```

File: sql-prepared-plan-cache.md [hunk 4]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
...
+
+```
+mysql> prepare st from 'select * from t where a > (select max(a) from t)';
+Query OK, 0 rows affected, 1 warning (0.01 sec)
+
+mysql> show warnings; -- 查询包含子查询无法被缓存
```
Comment (TomShawn) [inline sql-prepared-plan-cache.md:143]:
```suggestion
mysql> show warnings;  -- 查看查询计划无法被缓存的原因
```

File: sql-prepared-plan-cache.md [hunk 5]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
 1 row in set (0.00 sec)
 ```
 
+## Prepared Plan Cache 诊断
+
+对于无法进行缓存的查询或计划，TiDB 会通过 warning 的方式输出其无法被缓存的原因。
+
+```
+mysql> prepare st from 'select * from t where a > (select max(a) from t)';
```
Comment (TomShawn) [inline sql-prepared-plan-cache.md:140]:
```suggestion
mysql> prepare st from 'select * from t where a > (select max(a) from t)';  -- 该查询包含子查询，因此无法被缓存
```

File: sql-prepared-plan-cache.md [hunk 6]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
...
++---------+------+-----------------------------------------------+
+1 row in set (0.00 sec)
+
+mysql> prepare st from 'select * from t where a<?';
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> set @a='1'; 
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> execute st using @a;  -- 优化中进行了非 INT 类型到 INT 类型的转换，产生的计划可能随着参数变化有风险，因此不缓存
```
Comment (TomShawn) [inline sql-prepared-plan-cache.md:157]:
```suggestion
mysql> execute st using @a;  -- 该优化中进行了非 INT 类型到 INT 类型的转换，产生的执行计划可能随着参数变化而存在风险，因此 TiDB 不缓存该计划
```

**Review by TomShawn — APPROVED** [review]
建议大写 SQL 关键字

File: sql-prepared-plan-cache.md [hunk 7]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
 1 row in set (0.00 sec)
 ```
 
+## 诊断 Prepared Plan Cache
+
+对于无法进行缓存的查询或计划，可通过 `SHOW WARNINGS` 语句查看查询或计划是否被缓存。如果未被缓存，则可在结果中查看无法被缓存的原因。示例如下：
+
+```sql
+mysql> prepare st from 'select * from t where a > (select max(a) from t)';  -- 该查询包含子查询，因此无法被缓存
```
Comment (ran-huang) [inline sql-prepared-plan-cache.md:140]:
```suggestion
mysql> PREPARE st FROM 'SELECT * FROM t WHERE a > (SELECT MAX(a) FROM t)';  -- 该查询包含子查询，因此无法被缓存
```

File: sql-prepared-plan-cache.md [hunk 8]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
...
+Query OK, 0 rows affected, 1 warning (0.01 sec)
+
+mysql> show warnings;  -- 查看查询计划无法被缓存的原因
```
Comment (ran-huang) [inline sql-prepared-plan-cache.md:143]:
```suggestion
mysql> SHOW WARNINGS;  -- 查看查询计划无法被缓存的原因
```

File: sql-prepared-plan-cache.md [hunk 9]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
...
++---------+------+-----------------------------------------------+
+| Level   | Code | Message                                       |
++---------+------+-----------------------------------------------+
+| Warning | 1105 | skip plan-cache: sub-queries are un-cacheable |
++---------+------+-----------------------------------------------+
+1 row in set (0.00 sec)
+
+mysql> prepare st from 'select * from t where a<?';
```
Comment (ran-huang) [inline sql-prepared-plan-cache.md:151]:
```suggestion
mysql> PREPARE st FROM 'SELECT * FROM t WHERE a<?';
```

File: sql-prepared-plan-cache.md [hunk 10]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
...
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> set @a='1'; 
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> execute st using @a;  -- 该优化中进行了非 INT 类型到 INT 类型的转换，产生的执行计划可能随着参数变化而存在风险，因此 TiDB 不缓存该计划
```
Comment (ran-huang) [inline sql-prepared-plan-cache.md:157]:
```suggestion
mysql> EXECUTE st USING @a;  -- 该优化中进行了非 INT 类型到 INT 类型的转换，产生的执行计划可能随着参数变化而存在风险，因此 TiDB 不缓存该计划
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

File: sql-prepared-plan-cache.md [hunk 11]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
...
+| Level   | Code | Message                                       |
++---------+------+-----------------------------------------------+
+| Warning | 1105 | skip plan-cache: sub-queries are un-cacheable |
++---------+------+-----------------------------------------------+
+1 row in set (0.00 sec)
+
+mysql> PREPARE st FROM 'SELECT * FROM t WHERE a<?';
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> set @a='1';
```
Comment (TomShawn) [inline sql-prepared-plan-cache.md:154]:
```suggestion
mysql> SET @a='1'; 
```

File: sql-prepared-plan-cache.md [hunk 12]
```diff
@@ -132,6 +132,40 @@ MySQL [test]> select @@last_plan_from_cache;
...
+mysql> PREPARE st FROM 'SELECT * FROM t WHERE a<?';
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> set @a='1'; 
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> EXECUTE st USING @a;  -- 该优化中进行了非 INT 类型到 INT 类型的转换，产生的执行计划可能随着参数变化而存在风险，因此 TiDB 不缓存该计划
+Empty set, 1 warning (0.01 sec)
+
+mysql> show warnings;
```
Comment (TomShawn) [inline sql-prepared-plan-cache.md:160]:
```suggestion
mysql> SHOW WARNINGS;
```

**Pull request merged**